src/model/transaction/moveBlockInContentState.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule moveBlockInContentState
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+const invariant = require('invariant');
+
+import type ContentBlock from 'ContentBlock';
+import type ContentState from 'ContentState';
+import type {DraftInsertionType} from 'DraftInsertionType';
+
+function moveBlockInContentState(
+  contentState: ContentState,
+  blockToBeMoved: ContentBlock,
+  targetBlock: ContentBlock,
+  insertionMode: DraftInsertionType
+): ContentState {
+  invariant(
+    blockToBeMoved.getKey() !== targetBlock.getKey(),
+    'Block cannot be moved next to itself.'
+  );
+
+  invariant(
+    insertionMode !== 'replace',
+    'Replacing blocks is not supported.'
+  );
```

Comment:
Actually i am not really sure if i should implement the support for "replace" here, since at the moment there is no use case for it.

Actual use case:
- in "moveBlockInContentState" the insertionMode can be "before" or "after"
- in "AtomicBlockUtils" the insertionMode can be "before", "after" or "replace"

So i see 3 solutions:
1. implement "replace" in "moveBlockInContentState" even if there is no use case at the moment
2. implement 2 different types for "insertionMode", one to use in "moveBlockInContentState" and one to use in "AtomicBlockUtils"
3. we just leave it as it is at the moment

What do you think?